src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCaseEquivalences.cs

```diff
@@ -0,0 +1,165 @@
+                if (cultureName.Length < 2)
+                    return false;
+                else if (cultureName[0] == 't' && cultureName[1] == 'r' && (cultureName.Length == 2 || cultureName[2] == '-'))
+                    return true;
+                else if (cultureName[0] == 'a' && cultureName[1] == 'z' && (cultureName.Length == 2 || cultureName[2] == '-'))
+                    return true;
+                else
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// Performs a fast lookup which determines if a character is involved in case conversion, as well as
+        /// returns the characters that should be considered equivalent in case it does participate in case conversion.
+        /// </summary>
+        /// <param name="c">The character being analyzed</param>
+        /// <param name="equivalences">If <paramref name="c"/> is involved in case conversion, then equivalences will contain the
+        /// span of character which should be considered equal to <paramref name="c"/> in a case-insensitive comparison.</param>
+        /// <returns><see langword="true"/> if <paramref name="c"/> is involved in case conversion; otherwise, <see langword="false"/></returns>
+        /// <remarks>
+        /// The casing equivalence data is saved in three different lookup tables:
+        ///   EquivalenceFirstLevelLookup => This is a ushort array which contains an index to be used for searching on the next lookup table 'EquivalenceCasingMap'.
+        ///                                  We first grab the passed in <paramref name="c"/>, and divide it by 1024 and save it to index. We then use this index to
+        ///                                  perform a lookup in 'EquivalenceFirstLevelLookup' table. If the value at index is 0xFFFF, then <paramref name="c"/>
+        ///                                  isn't involved in case conversion so we keep equivalences as null, and return false. If the value at index is not 0xFFFF
+        ///                                  then we use that value to search in 'EquivalenceCasingMap'.
+        ///          EquivalenceCasingMap => This is a ushort array which contains a ushort for each character in a given range. The 3 highest bits of the ushort represent
+        ///                                  the number of characters that are considered equivalent to <paramref name="c"/>. The rest of the 13 bits of the ushort represent
+        ///                                  the index that should be used to get those equivalent characters in the 'EquivalenceCasingValues' table. We first calculate the
+        ///                                  index2 based on the value obtained from the first level lookup table and adding <paramref name="c"/> modulo 1024. If the value of
+        ///                                  EquivalnceCasingMap[index2] is 0xFFFF then <paramref name="c"/> isn't involved in case conversion so we return false. Otherwise,
```

Comment:
```suggestion
        ///                                  EquivalenceCasingMap[index2] is 0xFFFF then <paramref name="c"/> isn't involved in case conversion so we return false. Otherwise,
```